day_6/exercises/burrito.rb

```diff
@@ -11,9 +11,28 @@ def initialize(protein, base, toppings)
     @base     = base
     @toppings = toppings
   end
+
+  def add_topping(toppings)
+    "Please add #{toppings}."
+  end
+
+  def remove_topping(toppings)
+    "Please remove #{toppings}."
+  end
+
+  def change_protein(protein)
+    "I changed my mind I want #{protein}."
+  end
 end
 
 dinner = Burrito.new("Beans", "Rice", ["cheese", "salsa", "guacamole"])
+array = ["cheese", "salsa", "guacamole"]
```

Comment:
Have you tried passing the `array` variable to your declaration of the new Burrito instance?

Reply:
@mschae16

I didn't know how to add/remove elements from the array since the array wasn't defined on its own (the ["cheese", "salsa", "guacamole"] array was nested within the Burrito class). This is why on line 29 I defined a different array, being array = [<3 original toppings elements>]. Then, on lines 34 & 36, elements can be added and removed to this "array". 

I inserted "toppings = " into line 28 and removed line 29. Is there a better way to add/remove elements to this array or is this the best way? 

```
dinner = Burrito.new("Beans", "Rice", toppings = ["cheese", "salsa", "guacamole"])
#####array = ["cheese", "salsa", "guacamole"]
p dinner.protein
p dinner.base
p dinner.toppings
p dinner.add_topping("lettuce")
p toppings.push("lettuce")
```